### PR TITLE
Rename things with more friendly names

### LINKED ISSUE #26: Rename commands to remove PSA prefix and make them more user-friendly
### Summary
Some of the tool command names are chosen to match the opcode names in the wire protocol, but it would be better to give them names that are more intuitive for scripting, and also to cater for the fact that some commands might end up being compositions of multiple wire protocol ops. Also, words like "generate" and "destroy" are not idiomatic for scripting of key management: "create" and "delete" would be better. There is no need to align to PSA terminology here. The overriding aim is to make a nice usable CLI tool.

### Details
`psa-generate-key` should be renamed to either `create-key` or `create-rsa-keypair` depending on whether we want to specialise it to RSA in the future. A single `create-key` command will eventually need all sorts of differently-constrained parameter sets for all the different key types, so having a separate command for different main key types might make sense.

`psa-destroy-key` should be renamed to `delete-key` (and we only need one of those regardless of the key type).

`psa-generate-random` can just be `generate-random` or `make-random`.

`psa-export-key` can just be `export-key`.

`psa-export-public-key` can just be `export-publiic-key`.

### PR DESCRIPTION
**Pull request by hug-dev**
Fix #26 
It is based on PR #32 
Also start some work for #32 and #27 by creating the template files. They are obviously wrong now but can be changed when tackling those issues.

(just focus on the second commit for the review)

### CODE REVIEW

File: src/subcommands/delete_key.rs [hunk 1]
```diff
@@ -16,27 +16,25 @@ use structopt::StructOpt;
 /// Destroys a key.
 #[derive(Debug, StructOpt)]
 #[structopt(name = "destroy-key")]
```
Comment (hug-dev) [inline src/subcommands/delete_key.rs:18]: ah I need also to remove those lines everywhere... they are not used

File: src/subcommands/create_ecc_key.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Create an ECC key.
+//!
+//! Currently a lot of the parameters of the key generation are hardcoded because it is not clear
+//! on how it will be possible in the future to generate a key from the command line. This is
+//! currently useful for playing with the tool, and demonstrating the use of Parsec.
+//!
+//! This will generate a 2048 bits RSA key pair for signing.
```
Comment (ionut-arm) [inline src/subcommands/create_ecc_key.rs:10]: Will it, though?

File: src/subcommands/create_ecc_key.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Contributors to the Parsec project.
```
Comment (ionut-arm) [inline src/subcommands/create_ecc_key.rs:1]:
```suggestion
// Copyright 2021 Contributors to the Parsec project.
```

File: src/subcommands/create_ecc_key.rs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    bits: 2048,
+                    policy: Policy {
+                        usage_flags: UsageFlags {
+                            sign_hash: true,
+                            ..Default::default()
+                        },
+                        permitted_algorithms: AsymmetricSignature::RsaPkcs1v15Sign {
+                            hash_alg: Hash::Sha256.into(),
+                        }
+                        .into(),
```
Comment (ionut-arm) [inline src/subcommands/create_ecc_key.rs:60]: Copypasta

File: src/subcommands/create_ecc_key.rs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        Ok(NativeOperation::PsaGenerateKey(
+            psa_generate_key::Operation {
+                key_name: psa_generate_key_subcommand.key_name.clone(),
+                attributes: Attributes {
+                    lifetime: Lifetime::Persistent,
+                    key_type: Type::RsaKeyPair,
+                    bits: 2048,
+                    policy: Policy {
+                        usage_flags: UsageFlags {
+                            sign_hash: true,
```
Comment (ionut-arm) [inline src/subcommands/create_ecc_key.rs:54]: That should probs include `verify_hash` too?

File: src/subcommands/create_ecc_key.rs (hunk 2 above)
Comment (hug-dev) [reply src/subcommands/create_ecc_key.rs:10]: Yeah the text and content of all of the new files is wrong, planning to change it in a subsequent PR

File: src/subcommands/create_ecc_key.rs (hunk 5 above)
Comment (hug-dev) [reply src/subcommands/create_ecc_key.rs:54]: Yes will update all of that after

**Comment by hug-dev** [comment]
Rebased and modified copyrights. The docs and content of the ECC/encrypt/decrypt/sign/verify is currently all copy-paste but this PR is only to create files/rename to make subsequent PRs easier.

**Review by ionut-arm — APPROVED** [review]
I'm not convinced having all the different encrypt operations under one name is ideal, but we can debate this further

**Pull request merged**